p2p/book/book.go

```diff
@@ -0,0 +1,434 @@
+package book
+
+import (
+	"bufio"
+	"container/list"
+	"encoding/json"
+	"fmt"
+	"hash/crc64"
+	"io"
+	"math/rand"
+	"sort"
+	"strconv"
+	"sync"
+	"time"
+
+	ma "github.com/multiformats/go-multiaddr"
+	manet "github.com/multiformats/go-multiaddr/net"
+)
+
+const SELF ID = "SELF"
+
+type ID = string
+
+type Address = ma.Multiaddr
+
+type class uint32
+
+const (
+	deleted class = iota
+	stale
+	learned
+	stable
```

Comment:
i don't know what useful i can add, it is private classes. developers are expected to read code to understand what they are about

Reply:
the condition to put an address in the class would help. but i don't insist.

implementation comments often helps reader to understand faster. yes i will eventually understand it but it'll take longer. i guess this is a natural entry barrier :D

Reply:
I'd also be in favour of commenting these for two reasons.

* As @countvonzero said, it takes time to understand what these definitions mean, there are 40 different references to these classes throughout the diff, so understanding really what they do is not quick or easy. If a quick comment can save all that time then as a team it's much more efficient to comment than to not comment.
* The idea that you just need to read the code to understand what it does assumes that the code is correct. If there is a bug and the written code does not actually follow the author's original intent, the readers may never know because there was no comment to let them understand the original intent of the code.

Reply:
the intent is expressed in tests